selfservice/strategy/handler.go

```diff
@@ -0,0 +1,25 @@
+package strategy
+
+import (
+	"github.com/julienschmidt/httprouter"
+	"github.com/ory/herodot"
+	"github.com/ory/kratos/driver/config"
+	"github.com/ory/kratos/x"
+	"net/http"
+)
+
+const EndpointDisabledMessage = "This endpoint was disabled by system administrator. Please check your url or contact the system administrator to enable it."
+
+func IsDisabled(c interface {
+	config.Provider
+	x.WriterProvider
+}, strategy string, wrap httprouter.Handle) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+		enabled := c.Config(r.Context()).SelfServiceStrategy(strategy).Enabled
+		if !enabled {
+			c.Writer().WriteError(w, r, herodot.ErrBadRequest.WithError(EndpointDisabledMessage))
```

Comment:
Let's make this `herodot.ErrNotFound.WithDescription(EndpointDisabledMessage)` :)

Reply:
OK

Reply:
There is no method named `WithDescription` in `DefaultError`. I think it should be `WithReason`?